core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -85,7 +89,7 @@ public static void buildBroadcastShardsHeader(XContentBuilder builder, ToXConten
         builder.endObject();
     }
 
-    public static QuerySourceBuilder parseQuerySource(RestRequest request) {
+    public static QueryBuilder<?> parseQuerySource(RestRequest request) {
```

Comment:
I agree that it should be renamed but what do you suggest as a name? The method is taking the query params from the URL and converting them into a queryBuilder.